src/cluster.c

```diff
@@ -3389,10 +3412,10 @@ void clusterSetGossipEntry(clusterMsg *hdr, int i, clusterNode *n) {
     gossip->ping_sent = htonl(n->ping_sent/1000);
     gossip->pong_received = htonl(n->pong_received/1000);
     memcpy(gossip->ip,n->ip,sizeof(n->ip));
-    gossip->port = htons(n->port);
+    gossip->pport = htons(n->tcp_port);
     gossip->cport = htons(n->cport);
     gossip->flags = htons(n->flags);
-    gossip->pport = htons(n->pport);
+    gossip->port = htons(n->tls_port);
```

Comment:
In this gossip message, port is TLS and pport is TCP, regardless of cluster bus TLS, and the CLUSTERMSG_FLAG0_FIXED_PORT flag is set to indicate this.

But an old Redis version that doesn't understand the new CLUSTERMSG_FLAG0_FIXED_PORT flag receives this gossip, I think there is a problem. How are you thinking?

Reply:
The key is that Redis now uses "tls-cluster" to determine what `port` in a gossip message represents, and as long as we want to get rid of this logic, there is bound to be a compatibility problem. I don't think this logic is reasonable, so I believe we should try to solve the compatibility problem.

I don't think someone will deliberately run different versions of Redis in the same cluster, so we can assume the problem only happens during rolling upgarde: when tls-cluster is disabled, old version Redis might provide wrong port to non-TLS clients before upgraded. 

> set CLUSTERMSG_FLAG0_FIXED_PORT to 0 in Redis 8 and only in Redis 9 we set it to 1?

I don't want to take two big versions to solve this... If it is not tolerable, we may have a dynamic config "gossip-compatible-mode" to control whether new gossip protocol is used.

Reply:
I believe we need to support mixed versions in a cluster, but I'm not sure. @madolson WDYT?

I tried to find info on the redis.io website but it doesn't mention this in https://redis.io/docs/about/releases/#support nor in the cluster pages. I may have missed it. It would be good to have these interoperability rules documented, including for replication RDB/AOF compatibility.